### PR TITLE
middleware/fastcgi: Stripping PATH_INFO from SCRIPT_NAME

### PR DESCRIPTION
**Pull request by DenBeke**
While trying to get ownCloud to work I discovered that ownCloud produced a redirect loop once someone logged in. Digging through the source code of ownCloud I found out that this issue was related to the way ownCloud forms the urls parts that are passed to their router.

In their `getRawPathInfo` function they make use of `SCRIPT_NAME` environmental variable. When I compared this with Apache, I noticed that Caddy's `SCRIPT_NAME` included the `PATH_INFO` variable.  
https://github.com/owncloud/core/blob/7ff12d74d9f408836f1a40475e92723c60cf56d5/lib/private/appframework/http/request.php#L568
```php
/**
 * Get raw PathInfo from request (not urldecoded)
 * @throws \Exception
 * @return string Path info
 */
public function getRawPathInfo() {
    // ...
}
```

More info about the `SCRIPT_NAME` variable can be found in the CGI version 1.1 standard, [RFC 3875](https://www.ietf.org/rfc/rfc3875)

>  4.1.13.  SCRIPT_NAME
>  
> The SCRIPT_NAME string forms some leading part of the path component
>  of the Script-URI derived in some implementation-defined manner.  No
>  PATH_INFO segment (see section 4.1.5) is included in the SCRIPT_NAME
>  value.

This pull-request is the most naive patch I could come up with, and might not be totally like you want it. So tell me if I need to change something.

### CODE REVIEW

**Comment by mholt** [comment]
cf. #172 (related to fix_pathinfo)

File: middleware/fastcgi/fastcgi.go [hunk 1]
```diff
@@ -166,6 +166,12 @@ func (h Handler) buildEnv(r *http.Request, rule Rule, fpath string) (map[string]
 		scriptFilename = absPath
 	}
 
+	// Strip PATH_INFO from SCRIPT_NAME
+	indexPathInfo := strings.LastIndex(scriptName, pathInfo)
+	if indexPathInfo != -1 {
+		scriptName = scriptName[:indexPathInfo]
+	}
+
```
Comment (mholt) [inline middleware/fastcgi/fastcgi.go]: Maybe we want to use [strings.TrimPrefix](http://golang.org/pkg/strings/#TrimPrefix) here.

**Comment by abiosoft** [comment]
Hi, how about we do that on this line https://github.com/DenBeke/caddy/blob/master/middleware/fastcgi/fastcgi.go#L165. ? 

That basically means `SCRIPT_NAME` will hold same value as `DOCUMENT_URI`

**Comment by DenBeke** [comment]
Will look into it once I have ownCloud running. That way I can it a bit more...

**Comment by mholt** [comment]
Alright, I'm ready to merge this in, although I want to replace the LastIndex stuff with TrimPrefix, I think. Also the PR needs to be rebased with the latest master. Also, @DenBeke, make any changes you'd like (if any) related to @abiosoft's suggestion - if it works, of course ;)

**Comment by DenBeke** [comment]
Will do, @mholt.

**Comment by DenBeke** [comment]
It should be alright now, @mholt...

**Pull request merged**